irr/src/CGUIEditBox.cpp

```diff
@@ -1107,7 +1107,7 @@ void CGUIEditBox::breakText()
 				// TODO: I (Michael) think that we shouldn't change the text given by the user for whatever reason.
 				// Instead rework the cursor positioning to be able to handle this (but not in stable release
 				// branch as users might already expect this behavior).
-				Text.erase(i + 1);
+				Text.erase(i);
```

Comment:
This should erase the \r at `i`, and leave the string with just \n (that was in `i+1`